File: app/src/main/java/com/imperium/academio/ui/model/ClassRegisterRvModel.java
package com.imperium.academio.ui.model;

import androidx.annotation.Nullable;

public class ClassRegisterRvModel {
    private final String name;
    private final String key;

    public ClassRegisterRvModel(String key, String name) {
        this.key = key;
        this.name = name;
    }

    public String getKey() {
        return key;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (!(obj instanceof ClassRegisterRvModel)) {
            return false;
        }
        return ((ClassRegisterRvModel) obj).key.equals(key) && ((ClassRegisterRvModel) obj).name.equals(name);
    }
}

File: app/src/main/java/com/imperium/academio/ui/model/ClassRegisterRvModel.kt
package com.imperium.academio.ui.model

class ClassRegisterRvModel(val key: String?, val name: String?) {
    override fun equals(other: Any?): Boolean {
        return if (other !is ClassRegisterRvModel) {
            false
        } else other.key == key && other.name == name
    }

    override fun hashCode(): Int {
        var result = key.hashCode()
        result = 31 * result + name.hashCode()
        return result
    }
}